### ISSUE TITLE
https://[cname_domain]/hubpress doesn't work

### ISSUE THREAD

**Issue by icarito** [HubPress/hubpress.io#442]
https://[cname_domain]/hubpress for custom domain-hosted (CNAME) blogs.

The note that appears (following) isn't clear, and doesn't provide a solution. I can't enter the administration console when using a [cname domain]

*"If you are using a CNAME, you might need to access your Administration console using the https://<username>.github.io address pattern initially, so your CNAME redirect will work."*

**Comment by icarito** [comment]
In my case it's trying to open a 404, and looks to be unig [repositoryname] where it shouldn't:

http://centroslibres.org/centroslibres/themes/casper/theme.json?dt=1475226009017

Here's my source https://github.com/somosazucar/centroslibres

**Comment by icarito** [comment]
This is the same as https://github.com/HubPress/hubpress.io/issues/244 I guess

**Comment by jaredmorgs** [comment]
I'll check this in the docs next time I'm doing a bug fix run.

**Comment by icarito** [comment]
Thanks! I loved hubpress but I haven't been able to make it work reliably.

**Comment by jasonbullard** [comment]
I was having the same issue and discovered that you can add the CNAME to config.json and everything seems to work.

`
{
  "meta": {
   "username": "username or organization",
   "repositoryName": "blog",
    "cname": "blog.example.com",
    "branch": "gh-pages"
  },
  "theme": {
    "name": "Casper"
  }
}`

**Comment by icarito** [comment]
Thanks! That solved the issue for me! Would be nice to make a findable comment in documentation.

**Comment by jaredmorgs** [comment]
Yep, I'll look at making this more discoverable in the GitBook hosted docs.
-- 

Sent from Mobile.

**Comment by anthonny** [comment]
Hi,

@jasonbullard @icarito did you set the CNAME in the settings of your repository? 

HubPress do not work yet with this option, it still works with the CNAME file.

**Comment by icarito** [comment]
I can confirm that @ jasonbullard 's tip worked for me. Otherwise URLS
were not constructed correctly.

It would be ideal to only change this in one place. Or prominently
documented.

Also the program doesn't offer very friendly error messages...

**Comment by anthonny** [comment]
FYI, actually i can't set the `custom domain` of a repository with the Github API

See https://twitter.com/anthonny_q/status/783771728719835140

**Comment by jaredmorgs** [comment]
Next action:

1. Update https://hubpress.gitbooks.io/hubpress-knowledgebase/content/admin/access_admin_console.html with a better admonition.
2. Create an article about setting the CNAME in both places.
3. Note that GitHub can not set CNAME.
